dm/checker/checker.go

```diff
@@ -246,13 +248,37 @@ func (c *Checker) Init(ctx context.Context) (err error) {
 		}
 	}
 
-	if checkingShard {
-		for name, shardingSet := range sharding {
-			if shardingCounter[name] <= 1 {
-				continue
+	instance := c.instances[0]
+	if checkingShard && instance.cfg.Mode != config.ModeIncrement {
+		checkpointSQLs := []string{
+			fmt.Sprintf("SHOW CREATE TABLE %s", dbutil.TableName(instance.cfg.MetaSchema, cputil.LoaderCheckpoint(instance.cfg.Name))),
```

Comment:
seem this should impl in  `func (c *Checker) IsFreshTask() (bool, error) `

light suggestion, maybe do this job in next pr or left o todo item here